### ISSUE TITLE
Keys that autopair use should be able to perform their original actions after this plugin loaded

### ISSUE THREAD

**Issue by kutsan** [hlissner/zsh-autopair#14]
zsh-autopair overrides some existing keys such as `"`, `'` and such. Those keys can't be used after zsh-autopair sourced if users have existing widgets for that.

Recently, owner of this plugin added two fallback for `^?` (Backspace) and ` ` (Space) keys but apparently more general solution needed.

**Comment by hlissner** [comment]
Alright, a more generalized solution is in place. Let me know if it gives you any trouble.

**Comment by kutsan** [comment]
Commit a01f07411dc44fe94142fac30cf36418f8d86456 introducing a new error. It breaks the zsh-syntax-highlighting plugin and its some features are not working correctly. Everything is okay under b486f0dbdae4b420538e5fd5fe45f5f7c77316f6.

![](https://i.imgur.com/Yw4Yf6l.png)

**Comment by hlissner** [comment]
I found the problem and pushed a fix. Let me know if it gives you trouble!

**Comment by kutsan** [comment]
Well, I have still the same problem. I can confirm I am on e73053db1fef731e8001062b0a739dd20caed5fb.

**Comment by z0rc** [comment]
I don't want to derail the plugin development, but may the features be developed in separate branches, rather then in master?

**Comment by kutsan** [comment]
@z0rc I agree with that.

**Comment by hlissner** [comment]
Agreed. Sorry, should've done that from the start. I've reset master to https://github.com/hlissner/zsh-autopair/commit/b486f0dbdae4b420538e5fd5fe45f5f7c77316f6 and moved this feature to the `widget-fallback` branch.

**Comment by kutsan** [comment]
Also, I wouldn't force push if I were you. Now, probably someone dealing with their weird looking git history and wondering what went wrong. There is [safe](https://developer.atlassian.com/blog/2015/04/force-with-lease/force-with-lease.jpg) alternative for that.

**Comment by hlissner** [comment]
I apologize for any trouble, but I understand the pitfalls. Also, `--force-with-lease`'s selling point is it prevents overwriting "unpulled" commits on the remote. It wouldn't have made a difference here.

In any case, your feature is now in `widget-fallbacks`, could you try it out? If it doesn't work, would you also mind trying `develop` too? I haven't had much luck reproducing your past issues across several docker images and I'd like to be certain.

**Comment by kutsan** [comment]
Okay here's my report with this lines in my config.

```sh
bindkey -M viins '^?' backward-kill-word
source "$ZDOTDIR/plugins/zsh-autopair/autopair.zsh"
```

In `develop` branch, everything is like that. My binding works but there is nothing left of autopair. Backspace always deletes word backwards. Whereas it should have deleted the last pair if there is one.

![](https://i.imgur.com/LuLWKsO.gif)

In `widget-fallbacks` branch, everything still looks the same like I said above. It breaks zsh-syntax-highlighting and binding doesn't work at all. autopair and its other features works though.
